Add Mollie to DefaultPaymentGateways

samueljseay opened the pull request:
This adds the Mollie plugin to the set of DefaultPaymentGateways.

There is not a corresponding PR in the Mollie plugin for the auto config just yet.

As part of this I removed the jsx SVG for Mollie and put the svg in assets instead so the front-end/back-end can both reference an image path, since we can't define the rendering of a JSX component from the DefaultPaymentGateways config.

Part of woocommerce/woocommerce#32258 

### Detailed test instructions:

TBD
<!-- 
Please add your test instructions to `/TESTING-INSTRUCTIONS.md`.
-->

<!--- Please add a Changelog note

Enter a changelog note following the WooCommerce core format using prefixes of Enhancement:, Tweak:, Dev:, Fix:, Performance: to readme.txt under the "unreleased" list at the top of the changelog. If none exists, please add it. Also include PR number.

If changes pertain to a package, also update CHANGELOG.md in the package's folder in a similar manner.--->

samueljseay committed:
Add Mollie to DefaultPaymentGateways

This adds the Mollie plugin to the set of DefaultPaymentGateways.

There is not a corresponding PR in the Mollie plugin for the auto config just yet.

joshuatf commented on src/Features/RemotePaymentMethods/DefaultPaymentGateways.php:
```diff
@@ -58,6 +58,30 @@ public static function get_all() {
 				),
 			),
+			array(
+				'key'        => 'mollie',
```
I know we're not entirely sure on the path forward here, but maybe we should update this to `mollie_wc_gateway_banktransfer` or one of the other Mollie gateways so the payment gateway REST API doesn't 404.

We could probably even use this gateway to provide the help text needed for this gateway.

samueljseay replied:
Resolved here https://github.com/woocommerce/woocommerce-admin/pull/7092/commits/6557c4d2ba76c3bea7d40adf8999bb406c1c52f2

joshuatf commented on client/task-list/tasks/payments/methods/index.js:
```diff
@@ -342,7 +341,7 @@ export function getPaymentMethods( {
 			key: 'mollie',
 			title: __( 'Mollie Payments for WooCommerce', 'woocommerce-admin' ),
-			before: <MollieLogo />,
+			before: <img src={ wcAdminAssetUrl + 'mollie.svg' } alt="" />,
```
Good call updating this 👍

samueljseay replied:
Its weird the other images never had alts either , but it totally should :/ I'll add one

samueljseay replied:
I think we should be translating the alt too? I'm gonna add a translation call here, but yeah the other images will still be missing them. It seems out of scope to fix that here, but I could open another PR if its desired, seems maybe low value if this is going away.

samueljseay replied:
Resolved here https://github.com/woocommerce/woocommerce-admin/pull/7092/commits/809c8aa6c26c457f59b2b08887bc2f765376a5d7

joshuatf reviewed (approved):
Looks good!  Pre-approving pending an optional gateway key change but the WooCommerce Admin side of this is working well.

samueljseay committed:
Move Mollie logo to images/onboarding

samueljseay committed:
Update DefaultPaymentGateways.php

Use the bank transfer gateway for the fallback, we'll use this gateway to retrieve help text.

samueljseay committed:
Add a translated alt tag for the Mollie logo

samueljseay committed:
Add changelog entry

samueljseay committed:
Merge branch 'main' into add/mollie-default-gateway
